scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/codecs/package.scala

```diff
@@ -0,0 +1,102 @@
+package scala.meta.internal.semantic
+
+import org.scalameta.unreachable
+import scala.meta.io._
+import scala.meta.{semantic => m}
+import scala.meta.internal.semantic.{proto => p}
+
+package object codecs {
+  implicit class XtensionProtoSerializable[A](val a: A) extends AnyVal {
+    def toMeta[B](implicit ev: ProtoDecoder[B, A]): B = ev.fromProto(a)
+    def toProto[B](implicit ev: ProtoEncoder[A, B]): B = ev.toProto(a)
+    def toProtoOpt[B](implicit ev: ProtoEncoder[A, B]): Option[B] = Option(toProto[B])
+  }
+
+  implicit val AttributedSourceCodec: ProtoCodec[m.AttributedSource, p.AttributedSource] =
```

Comment:
Following discussions in https://github.com/scalameta/scalameta/issues/777 

We will need to have a a `sourceRoot: AbsolutePath` available in scope in order to turn `p.AttributedSource.path/relative_path: String` to construct an AbsolutePath.